apps/desktop-wallet/src/index.tsx

```diff
@@ -46,8 +47,10 @@ const queryClient = new QueryClient({
     queries: {
       staleTime: ONE_MINUTE_MS,
       retry: (failureCount, error) => {
-        // TODO: We should account for 429 errors from other libraries other than web3
-        if (!error.message.includes('Status code: 429')) {
+        if (
+          (error instanceof AxiosError && error.response?.status !== 429) ||
+          (error instanceof String && !error?.message?.includes('Status code: 429'))
```

Comment:
It'd be awesome if we had a `AlephiumApiError` or sth. Cheng proposed that we could extend the web3 package to allow a customizable error function, same way we allow for customizable fetch function. I would rather focus on the Tanstack task for now and handle the errors as we have been doing it so far (just a string)